src/main/java/org/purpurmc/purpurextras/modules/LoomMaxLayersModule.java

```diff
@@ -0,0 +1,128 @@
+    }
+
+    @Override
+    public boolean shouldEnable() {
+        return this.maxLayers != 6;
+    }
+
+    @EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = true)
+    public void onInventoryClick(InventoryClickEvent event) {
+        Player player = (Player) event.getWhoClicked();
+        ItemStack currentItem = event.getCurrentItem();
+        ItemStack cursorItem = event.getCursor();
+
+        if (!(event.getInventory() instanceof LoomInventory)) return;
+        if (currentItem != null && MaterialSetTag.BANNERS.isTagged(currentItem.getType())) {
+            if (newLayerExceedsMaxLayers(currentItem)) return;
+            switch (event.getSlotType()) {
+                case CRAFTING -> {
+                    if (isPlaceAction(event.getAction())) {
+                        save(player, currentItem);
+                    } else {
+                        undo(player, currentItem);
+                    }
+                }
+                case CONTAINER, QUICKBAR -> {
+                    if (event.getAction() == InventoryAction.MOVE_TO_OTHER_INVENTORY) {
+                        save(player, currentItem);
+                    }
+                }
+                case RESULT -> craftLoad(player, currentItem);
+            }
+        } else if (cursorItem != null && MaterialSetTag.BANNERS.isTagged(cursorItem.getType())) {
+            if (newLayerExceedsMaxLayers(cursorItem)) return;
+            if (event.getSlotType().equals(InventoryType.SlotType.CRAFTING) && isPlaceAction(event.getAction())) {
+                save(player, cursorItem);
+            }
+        }
+    }
+
+    private boolean isPlaceAction(InventoryAction action) {
+        return action == InventoryAction.PLACE_SOME ||
+                action == InventoryAction.PLACE_ALL ||
+                action == InventoryAction.PLACE_ONE;
+    }
+
+    private void save(Player player, ItemStack banner) {
+        BannerMeta meta = (BannerMeta) banner.getItemMeta();
+        List<Pattern> patterns = meta.getPatterns();
+        if (patterns.size() <= 5) return;
+        playerBannerData.remove(player);
+        playerBannerData.put(player, patterns);
+        changeStack(banner);
+    }
+
+    private void changeStack(ItemStack banner) {
+        BannerMeta meta = (BannerMeta) banner.getItemMeta();
+        List<Pattern> patterns = meta.getPatterns();
+        while (patterns.size() > 5) patterns.remove(0);
+        meta.setPatterns(patterns);
+        banner.setItemMeta(meta);
+    }
+
+    private void undo(Player player, ItemStack banner) {
+        if (!playerBannerData.containsKey(player)) return;
+        List<Pattern> oldPatterns = playerBannerData.get(player);
+        BannerMeta meta = (BannerMeta) banner.getItemMeta();
+
+        meta.setPatterns(oldPatterns);
+        banner.setItemMeta(meta);
+        playerBannerData.remove(player);
+    }
+
+    private void craftLoad(Player player, ItemStack banner) {
```

Comment:
arbitrary method name. I don't even know what's going here after reading through the method content